daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/TestMain.scala

```diff
@@ -0,0 +1,162 @@
+
+object TestMain {
+
+  def main(args: Array[String]): Unit = {
+
+    TestConfig.parse(args) match {
+      case None => sys.exit(1)
+      case Some(config) => {
+        val encodedDar: Dar[(PackageId, DamlLf.ArchivePayload)] =
+          DarReader().readArchiveFromFile(config.darPath).get
+        val dar: Dar[(PackageId, Package)] = encodedDar.map {
+          case (pkgId, pkgArchive) => Decode.readArchivePayload(pkgId, pkgArchive)
+        }
+
+        val applicationId = ApplicationId("Script Test")
+        val clientConfig = LedgerClientConfiguration(
+          applicationId = ApplicationId.unwrap(applicationId),
+          ledgerIdRequirement = LedgerIdRequirement("", enabled = false),
+          commandClient = CommandClientConfiguration.default,
+          sslContext = None
+        )
+        val timeProvider: TimeProvider =
+          config.timeProviderType match {
+            case TimeProviderType.Static => TimeProvider.Constant(Instant.EPOCH)
+            case TimeProviderType.WallClock => TimeProvider.UTC
+            case _ =>
+              throw new RuntimeException(s"Unexpected TimeProviderType: $config.timeProviderType")
+          }
+        val commandUpdater = new CommandUpdater(
+          timeProviderO = Some(timeProvider),
+          ttl = config.commandTtl,
+          overrideTtl = true)
+
+        val system: ActorSystem = ActorSystem("ScriptTest")
+        implicit val sequencer: ExecutionSequencerFactory =
+          new AkkaExecutionSequencerPool("ScriptTestPool")(system)
+        implicit val materializer: Materializer = Materializer(system)
+        implicit val ec: ExecutionContext = system.dispatcher
+
+        val runner = new Runner(dar, applicationId, commandUpdater)
+        val (participantParams, participantCleanup) = config.participantConfig match {
+          case Some(file) => {
+            val source = Source.fromFile(file)
+            val fileContent = try {
+              source.mkString
+            } finally {
+              source.close
+            }
+            val jsVal = fileContent.parseJson
+            import ParticipantsJsonProtocol._
+            (jsVal.convertTo[Participants[ApiParameters]], () => ())
+          }
+          case None =>
+            val (apiParameters, cleanup) = if (config.ledgerHost.isEmpty) {
+              val sandboxConfig = SandboxConfig.default.copy(
+                timeProviderType = config.timeProviderType
+              )
+              val sandbox = new SandboxServer(sandboxConfig)
+              val sandboxClosed = new AtomicBoolean(false)
+
+              def closeSandbox(): Unit = {
+                if (sandboxClosed.compareAndSet(false, true)) sandbox.close()
+              }
+
+              try Runtime.getRuntime.addShutdownHook(new Thread(() => closeSandbox()))
+              catch {
+                case NonFatal(t) =>
+                  //logger.error("Shutting down Sandbox application because of initialization error", t)
```

Comment:
Seems like a useful logging statement. Is there a reason why it is disabled?

Reply:
Good catch, I've enabled logging and uncommented this.